packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2371,15 +2395,23 @@ export class SnapController extends BaseController<
       const { iconPath } = manifest.result.source.location.npm;
       const svgIcon = iconPath ? await location.fetch(iconPath) : undefined;
 
-      const auxiliaryFiles = manifest.result.source.files
-        ? await Promise.all(
-            manifest.result.source.files.map(async (filePath) =>
-              location.fetch(filePath),
-            ),
-          )
-        : [];
+      const auxiliaryFiles = await getSnapFiles(
+        location,
+        manifest.result.source.files,
+      );
+
+      const localizationFiles = await getSnapFiles(
+        location,
+        manifest.result.source.locales,
+      );
 
-      const files = { manifest, sourceCode, svgIcon, auxiliaryFiles };
+      const files = {
+        manifest,
+        sourceCode,
+        svgIcon,
+        auxiliaryFiles,
+        localizationFiles: getValidatedLocalizationFiles(localizationFiles),
```

Comment:
Not sure if I am missing something. Can you point me to where we validate that the manifest can be translated? I think it should be both a CLI failure and an install failure if the manifest is trying to use translation but doesn't have the proper localization files for it